pkg/webhook/nodenetworkconfigurationpolicy/validation_test.go

```diff
@@ -55,21 +59,53 @@ var _ = Describe("NNCP Conditions Validation Admission Webhook", func() {
 		}),
 		Entry("current policy successfully configured", ValidationWebhookCase{
 			policy:           testPolicy,
-			currentPolicy:    p(policyconditions.SetPolicySuccess, ""),
+			currentPolicy:    p(allNodes, policyconditions.SetPolicySuccess, ""),
 			validationFn:     validatePolicyNotInProgressHook,
 			validationResult: []metav1.StatusCause{},
 		}),
 		Entry("current policy not matching", ValidationWebhookCase{
 			policy:           testPolicy,
-			currentPolicy:    p(policyconditions.SetPolicyNotMatching, ""),
+			currentPolicy:    p(allNodes, policyconditions.SetPolicyNotMatching, ""),
 			validationFn:     validatePolicyNotInProgressHook,
 			validationResult: []metav1.StatusCause{},
 		}),
 		Entry("current policy failed to configure", ValidationWebhookCase{
 			policy:           testPolicy,
-			currentPolicy:    p(policyconditions.SetPolicyFailedToConfigure, ""),
+			currentPolicy:    p(allNodes, policyconditions.SetPolicyFailedToConfigure, ""),
 			validationFn:     validatePolicyNotInProgressHook,
 			validationResult: []metav1.StatusCause{},
 		}),
+		Entry("policy has invalid node selector key", ValidationWebhookCase{
+			policy:       p(map[string]string{"bad key": "bar"}, policyconditions.SetPolicySuccess, ""),
+			validationFn: validatePolicyNodeSelector,
+			validationResult: []metav1.StatusCause{{
+				Type:    metav1.CauseTypeFieldValueInvalid,
+				Message: "invalid label key: \"bad key\": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')",
+				Field:   "spec.nodeSelector",
+			}},
+		}),
+		Entry("policy has node selector value with length beyond the limit", ValidationWebhookCase{
+			policy:       p(map[string]string{"kubernetes.io/hostname": "jumbo-frame-bridge-cnv-qe-infra-23.cnvqe2.lab.eng.rdu2.redhat.com"}, policyconditions.SetPolicySuccess, ""),
```

Comment:
Maybe a generic long FQDN would be better, wdyt?

Reply:
Done